mlrun/api/api/utils.py

```diff
@@ -470,6 +471,41 @@ def process_function_service_account(function):
     )
 
 
+def ensure_function_security_context(function, auth_info: mlrun.api.schemas.AuthInfo):
+    """
+    For iguazio we enforce that pods are running with the matching unix user id
+    (depending on keep/override mode) and nogroup group id.
+
+    keep - always use the user id of the user that triggered the 1st run
+    override - use the user id of the user requesting to submit the run
+    manual - security context is not auto applied
+    """
+
+    # if security context is not required
+    # security context is not yet supported with spark runtime
+    if (
+        mlrun.mlconf.function.spec.security_context.mode == "manual"
+        or mlrun.runtimes.RuntimeKinds.is_local_runtime(function.kind)
+        or function.kind == mlrun.runtimes.RuntimeKinds.spark
+        or mlrun.mlconf.httpdb.authentication.mode != "iguazio"
+    ):
+        return
+
+    # TODO: for old functions being triggered after upgrading mlrun - enrich with project owner uid
+    if (
+        mlrun.mlconf.function.spec.security_context.mode == "keep"
+        and function.spec.security_context is not None
+    ):
+        return
+
+    function: mlrun.runtimes.pod.KubeResource
+    nogroup_id = 65534
```

Comment:
Save this as a constant somewhere. (Maybe also add a comment explaining this is an unsigned `-1`)

Reply:
made it configurable as per offline discussion with @Tankilevitch and @liranbg